File: phase-1-arrays/LC128/code.cpp
#include <iostream>
#include <unordered_set>
#include <vector>
#include <algorithm>

using namespace std;

int longestConsecutive(const vector<int>& arr)
{
    unordered_set<int> numbers(arr.begin(), arr.end());

    int maxCount = 0;

    for (int num : numbers)
    {
        
        if (numbers.find(num - 1) == numbers.end())
        {
            int currentNum = num;
            int currentCount = 1;

            while (numbers.find(currentNum + 1) != numbers.end())
            {
                currentNum++;
                currentCount++;
            }

            maxCount = max(maxCount, currentCount);
        }
    }

    return maxCount;
}

int main()
{
    vector<int> arr = {100, 4, 200, 1, 3, 2};

    cout << longestConsecutive(arr) << endl;

    return 0;
}

File: phase-1-arrays/LC128/code.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int LongestConsecutive(int[] nums)
    {
        HashSet<int> numbers = new HashSet<int>(nums);

        int maxCount = 0;

        foreach (int num in numbers)
        {
            
            if (!numbers.Contains(num - 1))
            {
                int currentNum = num;
                int currentCount = 1;

                while (numbers.Contains(currentNum + 1))
                {
                    currentNum++;
                    currentCount++;
                }

                maxCount = Math.Max(maxCount, currentCount);
            }
        }

        return maxCount;
    }
}